niljs/src/utils/transaction.ts

```diff
@@ -0,0 +1,70 @@
+import type { PublicClient } from "../clients/PublicClient.js";
+import { bytesToHex } from "../encoding/fromBytes.js";
+import type { Hex } from "../types/Hex.js";
+import type { ProcessedReceipt } from "../types/IReceipt.js";
+import { waitTillCompleted } from "./receipt.js";
+
+/**
+ * Represents a transaction that can be awaited for completion
+ */
+export class TransactionResponse {
+  public readonly hash: Hex;
+  constructor(
+    private readonly tx: Uint8Array,
+    private readonly client: PublicClient,
+    private readonly shardId: number,
+  ) {
+    this.hash = bytesToHex(this.tx);
+  }
+
+  /**
+   * Returns the transaction hash
+   */
+  getHash(): Hex {
+    return this.hash;
+  }
+
+  /**
+   * Waits until the transaction is completed
+   */
+  async waitTillCompleted(options?: { waitTillMainShard?: boolean }): Promise<ProcessedReceipt[]> {
+    return await waitTillCompleted(this.client, this.hash, options);
+  }
+
+  /**
+   * Waits until the transaction is completed
+   *
+   * @notice same logic as waitTillCompleted, more compatible with ethers.js libraries
+   */
+  async wait(options?: { waitTillMainShard?: boolean }): Promise<ProcessedReceipt[]> {
+    return await waitTillCompleted(this.client, this.hash, options);
+  }
+
+  /**
+   * Returns the string representation of the transaction hash
+   */
+  toString(): Hex {
+    return this.hash;
+  }
+
+  /**
+   * Converts the transaction to a hex string for backward compatibility
+   */
+  toHexString(): Hex {
```

Comment:
We don't need two the same method. remove this. extended APi make it harder to support